internal/tracinginterceptor/interceptor.go

```diff
@@ -39,54 +45,184 @@ var (
 
 // Params defines the parameters for creating the Interceptor
 type Params struct {
-	// Tracer is used to propagate context and to generate spans
-	Tracer opentracing.Tracer
-	// Transport is the name of the transport, it decides the propagation format and propagation carrier
+	Tracer    opentracing.Tracer
 	Transport string
+	Logger    *zap.Logger
 }
 
 // Interceptor is the tracing interceptor for all RPC types.
-// It handles both tracing observability and context propagation using OpenTracing APIs.
 type Interceptor struct {
+	tracer            opentracing.Tracer
+	transport         string
+	propagationFormat opentracing.BuiltinFormat
+	log               *zap.Logger
+}
+
+// PropagationCarrier is an interface to combine both reader and writer interface
+type PropagationCarrier interface {
+	opentracing.TextMapReader
+	opentracing.TextMapWriter
 }
 
 // New constructs a tracing interceptor with the provided parameter.
 func New(p Params) *Interceptor {
-	return &Interceptor{}
+	m := &Interceptor{
+		tracer:            p.Tracer,
+		transport:         p.Transport,
+		propagationFormat: getPropagationFormat(p.Transport),
+		log:               p.Logger,
+	}
+	if m.tracer == nil {
+		m.tracer = opentracing.GlobalTracer()
+	}
+	return m
 }
 
 // Handle implements interceptor.UnaryInbound
 func (m *Interceptor) Handle(ctx context.Context, req *transport.Request, resw transport.ResponseWriter, h transport.UnaryHandler) error {
-	// TODO: implement
-	panic("implement me")
+	extendedWriter, ok := resw.(transport.ExtendedResponseWriter)
+	if !ok {
+		m.log.Debug("ResponseWriter does not implement ExtendedResponseWriter, proceeding without additional tracing")
+		return h.Handle(ctx, req, resw)
+	}
+
+	parentSpanCtx, _ := m.tracer.Extract(m.propagationFormat, getPropagationCarrier(req.Headers.Items(), req.Transport))
```

Comment:
Correct me if I am wrong - for outbound, `req.Transport` is set by the [firstoutboundmiddleware](https://github.com/yarpc/yarpc-go/blob/dev/internal/firstoutboundmiddleware/middleware.go#L74), so for the cases without yarpc dispatcher, the `req.Transport` will be empty.

Why not use `m.transport`?